pkg/autodiscovery/listeners/kubelet.go

```diff
@@ -296,12 +297,48 @@ func (l *KubeletListener) createService(entity string, pod *kubelet.Pod, firstRu
 	}
 
 	l.m.Lock()
+	defer l.m.Unlock()
+	old, found := l.services[entity]
+	if found && kubeletSvcEqual(old, &svc) {
+		log.Debugf("Received a duplicated kubelet service '%s'", svc.entity)
+		return
+	}
 	l.services[entity] = &svc
-	l.m.Unlock()
 
 	l.newService <- &svc
 }
 
+// kubeletSvcEqual returns false if one of the following fields aren't equal
+// - hosts
+// - ports
+// - ad identifiers
+// - check names
+func kubeletSvcEqual(first, second Service) bool {
+	hosts1, _ := first.GetHosts()
+	hosts2, _ := second.GetHosts()
+	if !reflect.DeepEqual(hosts1, hosts2) {
+		return false
+	}
+
+	ports1, _ := first.GetPorts()
+	ports2, _ := second.GetPorts()
+	if !reflect.DeepEqual(ports1, ports2) {
```

Comment:
are we sure that `DeepEqual` manage unsorted list? should we sort the lists before?

Reply:
same for `hosts1|2`

Reply:
they're always sorted (see line 290)

Reply:
hosts are in a map, should be good